core/platform/src/syscalls_impl.rs

```diff
@@ -1,17 +1,19 @@
 //! Implements `Syscalls` for all types that implement `RawSyscalls`.
 
-use crate::{return_variant, RawSyscalls, Syscalls, YieldType};
-
-impl<S: RawSyscalls> Syscalls for S {
+impl<S: crate::RawSyscalls> crate::Syscalls for S {
     // -------------------------------------------------------------------------
     // Yield
     // -------------------------------------------------------------------------
 
     fn yield_wait() {
-        Self::raw_yield(YieldType::Wait);
+        Self::raw_yield_wait();
     }
 
     fn yield_no_wait() -> bool {
-        Self::raw_yield(YieldType::NoWait) != return_variant::FAILURE.into()
+        let mut flag = core::mem::MaybeUninit::uninit();
```

Comment:
What value does MaybeUninit get you here? Seems like unnecessary danger for no benefit.